app/src/main/java/org/simple/clinic/patient/PatientRepository.kt

```diff
@@ -49,10 +50,22 @@ class PatientRepository @Inject constructor(
   private var ongoingNewPatientEntry: OngoingNewPatientEntry = OngoingNewPatientEntry()
 
   fun search(name: String): Observable<List<PatientSearchResult>> {
+    val timingTracker = OperationTimingTracker("Search Patient", utcClock)
```

Comment:
There are some IDE inspection warnings in `OperationTimingTracker`. Can you please resolve them?

Reply:
I was reading `OperationTimingTracker`'s documentation and found this:

> **Note:** Since an operation can happen across multiple threads, this class has been designed
> with thread-safety in mind. So it uses immutable maps for tracking operations, and atomic
> primitives where necessary.

Using immutable values doesn't entirely offer thread safety right? It's still possible for two threads to change a value at the same time and result in a race condition unless you synchronize the function calls. Does this make sense?

Reply:
Yes, that was intentional. It isn't crucial for us report these events to Heap, so I felt it was alright that some events might occasionally not get reported to Heap while we avoid the expense of thread synchronisation.

My intention in making these immutable was only to avoid concurrent modification exceptions  for things that don't affect the actual functioning of the app. Does that make sense?